submissions/evgeniy24/memory-pair-game/app.js

```diff
@@ -0,0 +1,81 @@
+const cards = document.querySelectorAll('.card');
+const win = document.querySelector('.win');
+
+let isRolledCard = false;
+let lockRoll = false;
+let firstCard, secondCard;
+let winCondition = 0;
+
+function rollCard() {
+    if (lockRoll) return;
+    if (this === firstCard) return;
+    this.classList.add('roll');
+
+    if (!isRolledCard) {
+        //first clicked Card!
+        isRolledCard = true;
+        firstCard = this;
+    } else {
+         //second clicked Card!
+        isRolledCard = false;
+        secondCard = this;
+        //check cards for match!
+        checkCardMatch();
+    }
+}
+
+function checkCardMatch() {
+    if (firstCard.dataset.name === secondCard.dataset.name) {
+        //Cards matched!
+        freezeCards();
+        checkWinCondition();  
+    } else {
+        //Cards didnt match!
+        cancelRollCards();
+    }
+}
+
+function freezeCards() {    
+        firstCard.classList.add('clear'); 
+        firstCard.removeEventListener('click', rollCard);
+        secondCard.classList.add('clear');
+        secondCard.removeEventListener('click', rollCard);
+        winCondition++;  
+        resetCards();
+}
+
+function cancelRollCards() {
+    lockRoll = true;
+    setTimeout(function() {
+        firstCard.classList.remove('roll');
+        secondCard.classList.remove('roll');
+        resetCards();
+        }, 400);   
+}
+
+function resetCards() {
+    isRolledCard = false;
+    lockRoll = false;
+    firstCard = null;
+    secondCard = null;
+}
+
+function checkWinCondition() {
+    if (winCondition === 6) {
+        setTimeout(function() {
+            win.classList.add('win_open');
+        }, 500)
+    };
+}
+
+(function resetOrderCards() {
```

Comment:
Why did you use [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) here?

Reply:
fixed! Pleasee see the result!